src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,13 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    // можна підставити Machine і все буде працювати
```

Comment:
The comment here suggests that using `Machine` would work, but according to the checklist, you should parameterize `MachineProducer` in such a way that any implementation not using `Machine` or its subtype could not be created. Ensure that the implementation strictly uses `Bulldozer` as intended.